### PR TITLE
[prim] use prim_buf inside prim_subreg

### PR DESCRIPTION
**Pull request by tjaychen**
This gives us a natual anchor point if it is ever required.
Ideally, we would directly swap the rtl code with prim_flop_en.

However, doing so creates complications for DV's backdoor loading,
as that does not work well across module hierarchies.

Instead we add anchor buffers both on the write enable and the output
of the flop, which are the two areas we would most likely be concerned
with.

This still gives us a close by anchor point and does not impact existin
DV.

### CODE REVIEW

File: hw/ip/prim/rtl/prim_alert_sender.sv [hunk 1]
```diff
@@ -346,19 +354,19 @@ module prim_alert_sender
   `ASSERT(AlertState0_A, alert_set_q === alert_state_o)
 
   if (IsFatal) begin : gen_fatal_assert
-    `ASSERT(AlertState1_A, alert_req_i |=> alert_state_o)
+    `ASSERT(AlertState1_A, alert_req |=> alert_state_o)
```
Comment (msfschaffner) [inline hw/ip/prim/rtl/prim_alert_sender.sv:357]: is there a reason why `alert_req_i` is replaced by its internally buffered version? in terms of coverage, it would be better to keep the primary inputs/outputs in these assertions...

**Review by msfschaffner — APPROVED** [review]
LGTM in general - just one comment RE the SVAs...

File: hw/ip/prim/rtl/prim_alert_sender.sv (hunk 1 above)
Comment (tjaychen) [reply hw/ip/prim/rtl/prim_alert_sender.sv:357]: I only did that because i moved all design / dv to look at the buffered version.
since it's an input, would the fpv tools not understand this..?

Comment (msfschaffner) [reply hw/ip/prim/rtl/prim_alert_sender.sv:357]: Technically this works - but it may create a coverage gap in FPV.

I.e., instead of covering the entire path `alert_req_i -> buf -> internal logic -> alert_state_o` these SVAs now only cover `buf -> internal logic -> alert_state_o`, which can affect the FPV coverage metrics...

**Comment by tjaychen** [comment]
@weicaiyang @cindychip i'm going to do this instead of touching the `always_ff` instance.

File: hw/ip/prim/rtl/prim_subreg.sv [hunk 2]
```diff
@@ -53,6 +53,14 @@ module prim_subreg
     end
   end
 
+  logic wr_en_buf;
```
Comment (vogelpi) [inline hw/ip/prim/rtl/prim_subreg.sv:56]: Sorry for being late @tjaychen . It looks like the `wr_en_buf` is not used. Or is this related to DV backdoor loading? In this case we should probably add an `unused_` signal as this generates lint errors in all IPs.

Comment (tjaychen) [reply hw/ip/prim/rtl/prim_subreg.sv:56]: o shoot i completely missed that. sorry i'll fix.
